### DISCUSSION TITLE
OSError: headers already sent

### DISCUSSION THREAD

**Discussion by yuanjiushao** [miguelgrinberg/python-socketio#1283] [Q&A]
python-engineio=4.8.0
python-socketio=5.10.0
When the client disconnects, the server always encounters this exception:

Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/engineio/middleware.py", line 63, in __call__
    return self.engineio_app.handle_request(environ, start_response)
  File "/usr/local/lib/python3.10/site-packages/socketio/server.py", line 428, in handle_request
    return self.eio.handle_request(environ, start_response)
  File "/usr/local/lib/python3.10/site-packages/engineio/server.py", line 321, in handle_request
    start_response(r['status'], r['headers'] + cors_headers)
OSError: headers already sent

**Comment by miguelgrinberg** [comment]
What web server are you using? What is the output of `pip freeze`?

**Reply by yuanjiushao** [reply]
My Django project is deployed using Uwsgi+Nginx.
pip freeze:
aiosignal==1.3.1
alibabacloud-credentials==0.3.2
alibabacloud-dysmsapi20170525==2.0.24
alibabacloud-endpoint-util==0.0.3
alibabacloud-gateway-spi==0.0.1
alibabacloud-openapi-util==0.2.1
alibabacloud-tea==0.3.3
alibabacloud-tea-openapi==0.3.7
alibabacloud-tea-util==0.3.11
alibabacloud-tea-xml==0.0.2
APScheduler==3.10.1
asgiref==3.7.2
async-timeout==4.0.3
attrs==23.1.0
bcrypt==4.1.1
bidict==0.22.1
certifi==2023.11.17
cffi==1.16.0
charset-normalizer==3.3.2
coreapi==2.3.3
coreschema==0.0.4
cryptography==41.0.7
Django==4.2.8
django-apscheduler==0.6.2
django-cors-headers==4.3.1
django-redis==5.2.0
django-simpleui==2023.11.16
djangorestframework==3.14.0
djangorestframework-simplejwt==5.3.1
drf-api-tracking==1.8.4
drf-flex-fields==1.0.2
drf-yasg==1.21.7
et-xmlfile==1.1.0
frozenlist==1.4.0
h11==0.14.0
idna==3.6
inflection==0.5.1
itypes==1.2.0
Jinja2==3.1.2
jpush==3.3.9
jsonpath==0.82
jsonschema==4.19.1
jsonschema-specifications==2023.7.1
Markdown==3.5.1
markdown-it-py==3.0.0
MarkupSafe==2.1.1
mdurl==0.1.2
multidict==6.0.4
openpyxl==3.1.2
packaging==23.2
Pillow==10.1.0
pycparser==2.21
Pygments==2.17.2
PyJWT==2.8.0
pymssql==2.2.8
PyMySQL==1.1.0
pypinyin==0.49.0
python-engineio==4.8.0
python-socketio==5.10.0
pytz==2022.5
pytz-deprecation-shim==0.1.0.post0
PyYAML==6.0
redis==5.0.1
referencing==0.30.2
requests==2.31.0
rich==13.7.0
rpds-py==0.13.2
simple-websocket==1.0.0
six==1.16.0
sqlparse==0.4.3
swagger-spec-validator==3.0.3
typing_extensions==4.8.0
tzdata==2023.3
tzlocal==5.2
uritemplate==4.1.1
urllib3==2.1.0
wsproto==1.2.0
yarl==1.9.2

**Reply by miguelgrinberg** [reply]
Thanks. Yes, this has already been reported in https://github.com/miguelgrinberg/Flask-SocketIO/issues/2017 and I will be looking into it soon. You can try downgrading to the following versions in the meantime:

python-engineio==4.7.1
python-socketio==5.9.0
